ale_linters/cpp/ccls.vim

```diff
@@ -0,0 +1,32 @@
+" Author: Ye Jingchen <ye.jingchen@gmail.com>, Ben Falconer <ben@falconers.me.uk>, jtalowell <jtalowell@protonmail.com>
+" Description: A language server for C++
+
+call ale#Set('cpp_ccls_executable', 'ccls')
+call ale#Set('cpp_ccls_init_options', {})
+
+function! ale_linters#cpp#ccls#GetProjectRoot(buffer) abort
+    let l:project_root = ale#path#FindNearestFile(a:buffer, '.ccls-root')
+
+    if empty(l:project_root)
+        let l:project_root = ale#path#FindNearestFile(a:buffer, 'compile_commands.json')
+    endif
+
+    if empty(l:project_root)
+        let l:project_root = ale#path#FindNearestFile(a:buffer, '.ccls')
+    endif
+
+    return !empty(l:project_root) ? fnamemodify(l:project_root, ':h') : ''
+endfunction
```

Comment:
Could you move this function to `autoload/ale/handlers/ccls.vim` and use the the same function in all of the files?

Reply:
Moved those functions to `autoload/ale/handlers/ccls.vim`. Should tests also be adjusted?